datafusion/common/src/config.rs

```diff
@@ -971,29 +979,46 @@ impl<F: ConfigField + Default> ConfigField for Option<F> {
     }
 }
 
+fn parse<T>(input: &str) -> Result<T>
```

Comment:
- let's add `default` to the name to indicate how it's used
- let's decide whether we call the thing "parse" or "transform" and use the naming consistently


```suggestion
fn default_transform<T>(input: &str) -> Result<T>
```